### PR TITLE
rcv: Enhance parsing and validation for DocumentoReferencias

### PR DESCRIPTION
**Pull request by svillegas-cdd**
- Updated data models to support `folio_documento_referencia` as `int | str`.
- Added validation logic for handling alphanumeric and non-numeric folio values.
- Fixed edge cases and adjusted tests to reflect changes in parsing and validation.

Ref: https://app.shortcut.com/cordada/story/20539/

### CODE REVIEW

File: src/cl_sii/rcv/data_models.py [hunk 1]
```diff
@@ -121,7 +121,13 @@ class OtrosImpuestos(TypedDict):
     """
 
 
-class DocumentoReferencia(TypedDict):
+@pydantic.dataclasses.dataclass(
+    frozen=True,
+    config=pydantic.ConfigDict(
+        arbitrary_types_allowed=True,
+    ),
```
Comment (jtrobles-cdd) [inline src/cl_sii/rcv/data_models.py]: There are no arbitrary types in this dataclass.

```suggestion
```

File: src/cl_sii/rcv/data_models.py
Comment (jtrobles-cdd) [inline src/cl_sii/rcv/data_models.py]: > Folio de documento que referencia. Debe ser cero sólo si el documento tiene encendido el Indicador de referencia global. Puede ser alfanumérico si se trata de un documento no tributario (rango del 800)

```suggestion
    folio_documento_referencia: int | str
```

File: src/cl_sii/rcv/data_models.py [hunk 2]
```diff
@@ -132,6 +138,13 @@ class DocumentoReferencia(TypedDict):
     Folio Docto. Referencia
     """
 
+    @pydantic.field_validator('folio_documento_referencia')
+    @classmethod
+    def validate_folio_documento_referencia(cls, v: object) -> object:
+        if isinstance(v, int):
+            cl_sii.dte.data_models.validate_dte_folio(v)
+        return v
```
Comment (jtrobles-cdd) [inline src/cl_sii/rcv/data_models.py]: > Folio de documento que referencia. Debe ser cero sólo si el documento tiene encendido el Indicador de referencia global. Puede ser alfanumérico si se trata de un documento no tributario (rango del 800)

```suggestion
    def validate_folio_documento_referencia(cls, v: int | str) -> int | str:
        if isinstance(v, int):
            try:
                cl_sii.dte.data_models.validate_dte_folio(v) # Raises ValueError
            except ValueError:
                # If value is numeric but not a valid DTE folio, assume that the value represents some other kind of number (e.g. a purchase order number) and convert it to a string.
                v = str(v)
        elif isinstance(v, str):
            if v.strip().isdecimal():
                numeric_v = int(v)
                try:
                    cl_sii.dte.data_models.validate_dte_folio(numeric_v) # Raises ValueError
                except ValueError:
                    # If value is numeric but not a valid DTE folio, assume that the value represents some other kind of number (e.g. a purchase order number) and leave it as a string.
                    pass
                else:
                    v = numeric_v

        return v
```

**Review by jtrobles-cdd — CHANGES_REQUESTED** [review]

File: src/cl_sii/rcv/data_models.py
Comment (jtrobles-cdd) [inline src/cl_sii/rcv/data_models.py]: Assume that `folio_documento_referencia` is not a valid DTE folio if it's a string.

```suggestion
            if isinstance(folio_documento_referencia, str):
```

**Review by jtrobles-cdd — CHANGES_REQUESTED** [review]

**Comment by svillegas-cdd** [comment]
@jtrobles-cdd in the RCV parsing we have the function `_parse_rv_documento_referencias` https://github.com/cordada/lib-cl-sii-python/blob/develop/src/cl_sii/rcv/parse_csv.py#L1755-L1823

Which tries to convert all `folio_documento_referencia` into `int` values and if there is a value that is not a digit it raises a parsing error for the RCV, should we remove this validation?

**Comment by jtrobles-cdd** [comment]
> [...] in the RCV parsing we have the function `_parse_rv_documento_referencias` https://github.com/cordada/lib-cl-sii-python/blob/develop/src/cl_sii/rcv/parse_csv.py#L1755-L1823
> 
> Which tries to convert all `folio_documento_referencia` into `int` values and if there is a value that is not a digit it raises a parsing error for the RCV, should we remove this validation?

Yes, we should update that to allow for alphanumeric folios, while still handling multiple folios separated by `-`.

File: src/tests/test_rcv_parse_csv.py [hunk 3]
```diff
@@ -2579,24 +2815,26 @@ def test__parse_rv_documento_referencias(self) -> None:
...
                     'tipo_documento_referencia': 33,
-                    'folio_documento_referencia': 400,
+                    'folio_documento_referencia': '400',
                 },
             ]
             self.assertEqual(result, expected)
 
-        # Test case 8: Invalid folio - non-numeric
+        # Test case 8: Alphanumeric folio - valid
         with self.subTest("Invalid folio - non-numeric"):
```
[the remark this answers is not in the document]
Comment (jtrobles-cdd) [reply src/tests/test_rcv_parse_csv.py]: CC: @svillegas-cdd

[the remark this answers is not in the document]
Comment (svillegas-cdd) [reply src/tests/test_rcv_parse_csv.py]: Updated

**Review by jtrobles-cdd — APPROVED** [review]
